### PR TITLE
Add PHP 8 compatibility

### PR DESCRIPTION
**Pull request by alexander-schranz**

### CODE REVIEW

File: .github/workflows/test-application.yaml [hunk 1]
```diff
@@ -9,14 +9,43 @@ on:
...
+                      tools: 'composer:v2'
+                      env:
+                          SYMFONY_DEPRECATIONS_HELPER: weak
+
+                    - php-version: '8.0'
+                      lint: false
+                      coverage: false
+                      dependency-versions: 'highest'
+                      tools: 'composer:v2'
+                      composer-options: '--ignore-platform-reqs'
```
Comment (alexander-schranz) [inline .github/workflows/test-application.yaml]: should be removed

File: .github/workflows/test-application.yaml [hunk 2]
```diff
@@ -33,27 +65,31 @@ jobs:
...
 
             - name: Lint code
+              if: ${{ matrix.lint }}
               run: composer lint
 
             - name: Execute unit test cases
               run: composer test -- Tests/Unit --coverage-php var/coverage.php --coverage-html coverage-report --log-junit var/junit.xml
 
             - name: Execute functional test cases
+              if: ${{ matrix.functional }}
```
Comment (alexander-schranz) [inline .github/workflows/test-application.yaml]: There would be some refractoring needed to create snapshot for sulu 2.1 and sulu 2.2 so I did currently disable testing functional tests on php 7.3 as they currently all fails.

/cc @nnatter

Comment (niklasnatter) [reply .github/workflows/test-application.yaml]: think i would just increase the min sulu version for now 🙂

Comment (alexander-schranz) [reply .github/workflows/test-application.yaml]: Done!

File: Tests/Functional/Controller/responses/headless_website__test_index.json [hunk 3]
```diff
@@ -35,7 +35,7 @@
             "categories": [],
             "tags": [],
             "images": [],
-            "segments": [],
+            "segments": "@array@.optional()",
```
Comment (niklasnatter) [inline Tests/Functional/Controller/responses/headless_website__test_index.json]:
```suggestion
            "segments": "@array@.isEmpty().optional()",
```

File: Tests/Functional/Controller/responses/navigation__get.json [hunk 4]
```diff
@@ -13,6 +13,7 @@
                 "template": "default",
                 "parent": "@uuid@",
                 "url": "\/test-1",
+                "path": "@string@.optional()",
```
Comment (niklasnatter) [inline Tests/Functional/Controller/responses/navigation__get.json]: we could check the value of the path like this, but not sure if thats necessary 🙂 

```suggestion
                "path": "@string@.match(\"parent/child\").optional()",
```

File: Tests/Functional/Controller/responses/navigation__get_excerpt.json [hunk 5]
```diff
@@ -30,7 +31,7 @@
                     "categories": [],
                     "audience_targeting_groups": [],
                     "tags": [],
-                    "segments": [],
+                    "segments": "@array@.optional()",
```
Comment (niklasnatter) [inline Tests/Functional/Controller/responses/navigation__get_excerpt.json]:
```suggestion
                    "segments": "@array@.isEmpty().optional()",
```

File: Tests/Functional/Controller/responses/navigation__get_excerpt.json [hunk 6]
```diff
@@ -65,7 +67,7 @@
...
```
Comment (niklasnatter) [inline Tests/Functional/Controller/responses/navigation__get_excerpt.json]:
```suggestion
                    "segments": "@array@.isEmpty().optional()",
```

**Review by niklasnatter — APPROVED** [review]
nice to finally test the lowest dependencies here 🙂

**Comment by alexander-schranz** [comment]
@nnatter good suggestion like the solution 👍

**Pull request merged**